src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -729,6 +738,14 @@ namespace winrt::TerminalApp::implementation
             }
             _commandLineHistory.InsertAt(0, filteredCommand.value());
 
+            auto recentCommands = winrt::single_threaded_vector<hstring>();
+            for (const auto& recentFilteredCommand : _commandLineHistory)
+            {
+                recentCommands.Append(recentFilteredCommand.Item().Name());
+            }
+
+            ApplicationState::SharedInstance().RecentCommands(recentCommands);
```

Comment:
Will this cause problems if you have multiple windows? Seems like this would overwrite changes if you run a command in another window.

Reply:
Yeah. Good point. I guess reading the state close to the update will improve things. Otherwise, not sure if we can make it fully atomic without some IPC or cross-process critical section (which should probably be provided by a state itself).

Reply:
Reduced the potential race, by reading the state immediately before the update. There are better solutions for sure, but I am not sure what is the ROI of solving this, as the user is not expected to run command several windows within less than a second.